core/jobs/batch_jobs/exp_migration_jobs.py

```diff
@@ -117,8 +111,8 @@ def _generate_exploration_changes(
             exp_change = exp_domain.ExplorationChange({
                 'cmd': (
                     exp_domain.CMD_MIGRATE_STATES_SCHEMA_TO_LATEST_VERSION),
-                'from_version': exp_states_version,
-                'to_version': feconf.CURRENT_STATE_SCHEMA_VERSION
+                'from_version': str(exp_states_version),
```

Comment:
Why strings?

Reply:
Hm I don't know tbh, it sounds like it ought to be an int. Maybe check with @sahiljoster32?

Reply:
(And also @vojtechjelinek @srikanthkb since I think they wrote the exp migration job)

Reply:
I dug in a bit about this, it looks like from_version and to_version were represented / stored as strings until the v52 job was run. I manually checked the commit cmds stored for snapshots and noticed that previous migration job runs used strings while the latest migration changed it to int. We will need to do a clean up on the server to fix this so the data is consistent.

Reply:
Created https://github.com/oppia/oppia/issues/16383 to track. We need to decide on what type the version should actually be. I would say string because that was how it worked previously but open to suggestions.

Reply:
Looking at the current codebase state, I think it should be int but removing str from here https://github.com/oppia/oppia/blob/de52f5b11a4f9ae0d8ba2bbc526e148b88b81971/core/domain/exp_services.py#L926
Is throwing an error.

I think the measures that are taken by Kevin are correct ( and reasoning also ).

Why before str is added instead of int while adding annotations?
Because typing mostly relies on connecting dots between the return value, passed value, how value is used -- and here value is used as string not int.